### PR TITLE
Remove all mentions of "data files" and "garbage collection" remaining.

### PR DESCRIPTION
**Pull request by sebastiancarlos**
Also, introduce the concept of "working set" to explain how IDs work. Note that "garbage collection" is now called "rebuilding".

Fixes: #877

### CODE REVIEW

File: content/docs/3rd-party.md [hunk 1]
```diff
@@ -15,14 +15,11 @@ We encourage you to create such add-ons, but in doing so, there are some rules t
...
 - Produce, consume and handle UTF8 text properly.
   UTF8 is the only text encoding supported by Taskwarrior.
-- Don't attempt to parse the pending.data file.
-  Here's why: the `.data` file format is currently on its fourth version.
-  The very first version was never released, so if you want to read Taskwarrior data properly, you will need to parse the three supported formats.
-  Those formats are not documented.
-  Additionally, you will need to handle the GC operations, implement the task "unwait" feature, observe user defined attribute handling restrictions, and implement recurring task synthesis all of which require `.taskrc` and default value access.
+- Don't attempt to read the database.
+  Here's why: the database format might change in future versions.
+  Additionally, you will need to handle the rebuild operations, implement the task "unwait" feature, observe user defined attribute handling restrictions, and implement recurring task synthesis all of which require `.taskrc` and default value access.
```
Comment (djmitche) [inline content/docs/3rd-party.md:20]: Let's shorten this a little:
```suggestion
  Here's why: the database format might change in future versions, and there are important relationships
  between tables in the database which must be maintained, or tasks may be corrupted.
```

File: content/docs/3rd-party.md [hunk 2]
```diff
@@ -43,9 +40,8 @@ We encourage you to create such add-ons, but in doing so, there are some rules t
 
 ## Guidelines
 
-- If you need to store additional data, consider putting your own data file in the `~/.task` directory.
-  Just don't use the file names `pending.data`, `completed.data`, `backlog.data`, or `undo.data`.
+- If you need to store additional data, consider putting it in the `~/.task` directory.
 - There are many helper commands designed to assist add-on scripts such as shell completion scripts.
   These commands all begin with an underscore, see them with this command: `task help | grep ' _'`.
-- Familiarize yourself with the means of forcing color on or off, disabling word wrapping, disabling bulk operation limitations, disabling confirmation, disabling gc, modifying verbosity and so on.
+- Familiarize yourself with the means of forcing color on or off, disabling word wrapping, disabling bulk operation limitations, disabling confirmation, disabling rebuilding, modifying verbosity and so on.
```
Comment (djmitche) [inline content/docs/3rd-party.md:46]:
```suggestion
- Familiarize yourself with the means of forcing color on or off, disabling word wrapping, disabling bulk operation limitations, disabling confirmation, disabling rebuilding the working set, modifying verbosity and so on.
```

File: content/docs/ids.md [hunk 3]
```diff
@@ -4,29 +4,29 @@ title: "Taskwarrior - ID Numbers"
...
+The working set tends to stabilize at a constant size, regardless of the amount of work you are getting done [1].
+The total number of tasks, on the other hand, grows unbounded.
 This means that some reports, such as the `completed` or `all` report will run slower over time because there is more data to read and display.
 
 One nice side effect of this mechanism is that the ID numbers of pending tasks remain small, and less prone to error during data entry.
 
 One downside is that ID numbers change occasionally, although in a predictable way.
 
-## Garbage Collection
+## Rebuilding
```
Comment (djmitche) [inline content/docs/ids.md:22]:
```suggestion
## Rebuilding the Working Set
```

File: content/docs/ids.md [hunk 4]
```diff
@@ -4,29 +4,29 @@ title: "Taskwarrior - ID Numbers"
...
 When a task is marked as deleted, it gets a new status, and an `end` attribute, among other things.
-The task is written back into the `pending.data` file, but it doesn't belong there - it belongs in the `completed.data` file.
-There is an operation called Garbage Collection (gc) that is automatically run by taskwarrior to move tasks into the correct files.
-When moving tasks between the files, ID numbers are affected, because they are simply line numbers in the `pending.data` file.
+The task is still in the working set, but it doesn't belong there.
+There is an operation called "rebuilding" that is automatically run by taskwarrior to move tasks out of the working set.
+When moving tasks out of the working set, the ID numbers are changed, because they should be a sequence with no gaps assigned to all tasks in the working set.
 
-But gc is only run occasionally: it is run immediately before any command that displays ID numbers, such as a report, but not when a task is modified in any way.
+But rebuilding is only run occasionally: it is run immediately before any report that displays ID numbers, but not when a task is modified in any way.
```
Comment (djmitche) [inline content/docs/ids.md:29]:
```suggestion
But the working set is only rebuilt occasionally: immediately before any report that displays ID numbers, but not when a task is modified in any way.
```

File: content/docs/ids.md [hunk 5]
```diff
@@ -45,15 +45,14 @@ The next three commands use the ID numbers shown in the report.
...
 When the last command is run, new ID numbers are shown, and any subsequent commands would use those.
-This is why only commands that display ID numbers perform a gc first.
+This is why only reports that display ID numbers perform a rebuild first.
 
 ## Configuration
 
-The `gc` configuration setting may be set to 'off' to disable the gc operation.
-This has the effect of letting data accumulate in the `pending.data`
-file, and effectively making the ID numbers static.
+The `gc` configuration setting may be set to 'off' to disable the rebuild operation.
```
Comment (djmitche) [inline content/docs/ids.md:52]: I'm curious what your thoughts are here? The config uses `gc`, but we're trying to use "rebuild" everywhere else..

File: content/docs/terminology.md [hunk 6]
```diff
@@ -260,13 +260,13 @@ A filter is essentially an algebraic expression that must evaluate to`True` for
 Most commands accept filters.
 See [Filters](../filter/) for a more complete discussion.
 
-## gc
+## rebuilding
```
Comment (djmitche) [inline content/docs/terminology.md:263]: Maybe an "a.k.a." is good here?

**Review by djmitche — COMMENTED** [review]
This looks good! In general, I think "rebuild" is short for "rebuild the working set", and maybe we need to use the full phrase more often. I've added a few suggestions here but please apply your discretion :)

File: content/docs/ids.md (hunk 5 above)
Comment (sebastiancarlos) [reply content/docs/ids.md:52]: I would prefer to rename the config from `gc` to something like `rebuildworkingset`. But this depends on how you want to handle backwards compatibility.

File: content/docs/terminology.md (hunk 6 above)
Comment (djmitche) [reply content/docs/terminology.md:263]: If both terms "rc" and "rebuilding the working set' are used, then the docs should link them somewhere.

File: content/docs/ids.md (hunk 5 above)
Comment (djmitche) [reply content/docs/ids.md:52]: Let's leave this as-is, and please open a bug in GothenburgBitFactory/taskwarrior to rename the config option. We can keep the old one for backward compatibility.

Comment (sebastiancarlos) [reply content/docs/ids.md:52]: Sure! https://github.com/GothenburgBitFactory/taskwarrior/issues/3557

**Review by djmitche — APPROVED** [review]
Sorry for the delay. This is great - thanks!

**Pull request merged**